Firmware/PlatformIO/main_mcu/src/bms.cpp

```diff
@@ -19,6 +27,31 @@ void BMS::Tick(std::chrono::milliseconds elapsed_time)
     // todo
 }
 
+// Find maximum discharge and regen current
+void BMS::GetStateOfEnergy()
+{
+    // Update min voltage (max already set(?))
+    minVoltage = *std::min_element(voltages.begin(), voltages.end());
+
+    float currentPerCell = current[0] / kNumCellsParallel;
+    float invResistancePerCell = kNumCellsParallel / kInternalResistance;
+
+    // Find highest and lowest open cell voltage
+    float minOpenCircuitVoltage = minVoltage + (currentPerCell * kInternalResistance);
+    float maxOpenCircuitVoltage = maxVoltage + (currentPerCell * kInternalResistance);
+
+    // Limit at V_open + I * (R_internal / numCellsParallel) = V_boundary
+    //  => I = (numCellsParallel / R_internal) * (V_boundary - V_open)
+    maxDischargeCurrent = std::min(
+        invResistancePerCell * (kCellUndervoltage - minOpenCircuitVoltage),
```

Comment:
```suggestion
        invResistancePerCell * (minOpenCircuitVoltage - kCellUndervoltage),
```
Because undervoltage is less than the open circuit voltage (unless we're in an undervoltage fault)